src/microbe_stage/editor/upgrades/ChemoreceptorUpgradeGUI.cs

```diff
@@ -89,4 +89,25 @@ public void ApplyChanges(MicrobeEditor editor)
         storedOrganelle.SetCustomUpgradeObject(new ChemoreceptorUpgrades(shownChoices[compounds.Selected],
             (float)maximumDistance.Value, (float)minimumAmount.Value, colour.Color));
     }
+
+    public void CompoundChanged(int index)
+    {
+        if (shownChoices?[index] != null && shownChoices != null)
+        {
+            // If the color is in the shownChoices list don't change the color
+            bool isColorInCompundList = false;
+            foreach (Compound compound in shownChoices)
+            {
+                if (colour.Color == compound.Colour || colour.Color == Colors.White)
```

Comment:
Instead of having Colors.White here, I think it would be better to change the other place in this file where Colors.White is used to use the glucose colour, that way we shouldn't need the explicit check with white here.